bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/items/ItemUIRegistryImpl.java

```diff
@@ -543,24 +559,31 @@ private QuantityType<?> convertStateToWidgetUnit(QuantityType<?> quantityState,
         }
     }
 
-    private String getLabelFromWidget(Widget w) {
+    private WidgetLabelWithSource getLabelFromWidget(Widget w) {
         String label = null;
+        WidgetLabelSource source = WidgetLabelSource.NONE;
+
         if (w.getLabel() != null) {
             // if there is a label defined for the widget, use this
             label = w.getLabel();
+            source = WidgetLabelSource.SITEMAP_WIDGET;
         } else {
             String itemName = w.getItem();
             if (itemName != null) {
                 // check if any item ui provider provides a label for this item
                 label = getLabel(itemName);
                 // if there is no item ui provider saying anything, simply use the name as a label
-                if (label == null) {
+                if (label != null) {
+                    source = WidgetLabelSource.ITEM_LABEL;
+                } else {
                     label = itemName;
+                    source = WidgetLabelSource.ITEM_NAME;
                 }
             }
         }
         // use an empty string, if no label could be found
-        return label != null ? label : "";
+        String finalLabel = label != null ? label : "";
+        return new WidgetLabelWithSource(finalLabel, finalLabel.isEmpty() ? WidgetLabelSource.NONE : source);
```

Comment:
I am just annoyed with that because NONE in that case is not only the undefined source but also another source defined but being empty. Why not but in that case we need a clear comment in front of the enum for the NONE value.

Reply:
Or why not letting each UI checking if the label is empty ?
In that case, NONE will only have one meaning, related to the source and not to the value itself.

Reply:
I think the source should not carry information about the label value itself.

Reply:
So you basically just want me to return `source` here instead of the ternary? That's OK, I guess.

Reply:
Yes